lib/Froxlor/Validate/Validate.php

```diff
@@ -62,32 +62,77 @@ public static function validate($str, $fieldname, $pattern = '', $lng = '', $emp
-	 * @param string $lng
-	 *        	index for error-message (if $return_bool is false)
-	 * @param bool $allow_localhost
-	 *        	whether to allow 127.0.0.1
-	 * @param bool $allow_priv
-	 *        	whether to allow private network addresses
-	 * @param bool $allow_cidr
-	 *        	whether to allow CIDR values e.g. 10.10.10.10/16
-	 *        	
-	 * @return string|bool ip address on success, false on failure
-	 */
-	public static function validate_ip2($ip, $return_bool = false, $lng = 'invalidip', $allow_localhost = false, $allow_priv = false, $allow_cidr = false, $throw_exception = false)
+    /**
+     * Converts CIDR to a netmask address
+     *
+     * @thx to https://stackoverflow.com/a/5711080/3020926
+     * @param string $cidr
+     *
+     * @return string
+     */
+    public static function cidr2NetmaskAddr ($cidr) {
+
+        $ta = substr ($cidr, strpos ($cidr, '/') + 1) * 1;
+        $netmask = str_split (str_pad (str_pad ('', $ta, '1'), 32, '0'), 8);
+
+        foreach ($netmask as &$element) {
+            $element = bindec ($element);
+        }
+
+        return implode ('.', $netmask);
+    }
+
+    /**
+     * Checks if an $address (IP) is IPv6
+     *
+     * @param $address
+     *
+     * @return bool
+     * @thx https://stackoverflow.com/a/13677565/3020926
+     */
+    public static function is_ipv6($address) {
+        $ipv4_mapped_ipv6 = strpos($address, "::ffff:");
+        return (strpos($address, ":") !== FALSE) &&
+            ($ipv4_mapped_ipv6 === FALSE || $ipv4_mapped_ipv6 != 0);
+    }
+
+    /**
+     * Checks whether it is a valid ip
+     *
+     * @param string $ip
+     *            ip-address to check
+     * @param bool   $return_bool
+     *            whether to return bool or call \Froxlor\UI\Response::standard_error()
+     * @param string $lng
+     *            index for error-message (if $return_bool is false)
+     * @param bool   $allow_localhost
+     *            whether to allow 127.0.0.1
+     * @param bool   $allow_priv
+     *            whether to allow private network addresses
+     * @param bool   $allow_cidr
+     *            whether to allow CIDR values e.g. 10.10.10.10/16
+     * @param bool   $cidr_as_netmask
+     *            whether to format CIDR nodation to netmask notation
+     *
+     * @return string|bool ip address on success, false on failure
+     */
+	public static function validate_ip2($ip, $return_bool = false, $lng = 'invalidip', $allow_localhost = false, $allow_priv = false, $allow_cidr = false, $throw_exception = false, $cidr_as_netmask = false)
 	{
 		$cidr = "";
 		if ($allow_cidr) {
 			$org_ip = $ip;
 			$ip_cidr = explode("/", $ip);
-			if (count($ip_cidr) == 2) {
+			if (count($ip_cidr) === 2) {
+			    if (self::is_ipv6($ip)) {
+                    //MySQL does not handle CIDR of IPv6 addresses, return error
+                    if (false === filter_var($ip, FILTER_VALIDATE_IP, FILTER_FLAG_IPV6)) {
```

Comment:
please compare function-call to const/check-value, not the other way around, we never do that (as far as I'm aware)

Reply:
Changed.